ops/model.py

```diff
@@ -276,6 +276,9 @@ def get_secret(self, *, id: Optional[str] = None, label: Optional[str] = None) -
         """
         if not (id or label):
             raise TypeError('Must provide an id or label, or both')
+        if id is not None:
+            # Canonicalize to "secret:<id>" form for consistency in backend calls.
+            id = Secret._canonicalize_id(id)
```

Comment:
Do we want to keep the canonicalize_id call inside Secret.__init__ now that it's here? And replace it with a validation step? Or do you think a little bit of redundancy won't hurt?